### PR TITLE
Update the ASR finetuning tutorials with NeMo - Dropping TAO

### PR DESCRIPTION
**Pull request by VahidooX**
Updated the tutorial for fine-tuning ASR models with NeMo, and dropped TAO.

### CODE REVIEW

File: asr-python-finetuning-nemo.ipynb [hunk 1]
```diff
@@ -0,0 +1,3274 @@
...
+        "### ASR Model Export"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "ap5CjiV4_mMT"
+      },
+      "source": [
+        "With NeMo, you can also export your model in a format that can be deployed using NVIDIA RIVA; a highly performant application framework for multi-modal conversational AI services using GPUs. The same command for exporting to ONNX can be used here. The only small variation is the configuration for `export_format` in the spec file."
```
[the remark this answers is not in the document]
Comment (LynseyFabel) [reply asr-python-finetuning-nemo.ipynb:3103]: I think its odd that we're defining what Riva is at the end of the tutorial. This tutorial is about fine tuning a Riva model, so I would think we should define all related major software at the beginning of this topic.

File: asr-python-finetuning-nemo.ipynb [hunk 2]
```diff
@@ -0,0 +1,3274 @@
...
+        "## What's Next?"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "SJ7hWpRT_mMb"
+      },
+      "source": [
+        "You could use NeMo to build custom models for your own applications, and deploy them to Nvidia Riva! To try deploying these models to RIVA, use the text-to-speech-deployment.ipynb as a quick sample."
```
Comment (LynseyFabel) [inline asr-python-finetuning-nemo.ipynb:3244]:
```suggestion
        "You could use NeMo to build custom models for your own applications, and deploy them to NVIDIA Riva! To try deploying these models to Riva, use the `text-to-speech-deployment.ipynb` as a quick sample."
```

Comment (LynseyFabel) [reply asr-python-finetuning-nemo.ipynb:3244]: can you make text-to-speech-deployment.ipynb a link?

**Review by LynseyFabel — APPROVED** [review]
I've left a few comments. Let me know if you have any questions. My review is complete. Thank you!

File: asr-python-finetuning-nemo.ipynb [hunk 3]
```diff
@@ -0,0 +1,3274 @@
...
+          "name": "stdout",
+          "text": [
+            " Conformer-CTC-BPE.nemo\n",
+            "'Conformer-CTC-BPE--val_wer=1.3571-epoch=0.ckpt'\n",
+            "'Conformer-CTC-BPE--val_wer=1.3571-epoch=1-last.ckpt'\n"
+          ]
+        }
+      ]
+    },
+    {
```
Comment (svenchilton) [inline asr-python-finetuning-nemo.ipynb:2821]:
```suggestion
    {
      "cell_type": "code",
      "source": [
        "MODEL_PATH = os.path.join(DATA_DIR, 'checkpoints/Conformer-CTC-BPE/test/checkpoints/Conformer-CTC-BPE.nemo')"
      ],
    },
    {
```

File: asr-python-finetuning-nemo.ipynb [hunk 4]
```diff
@@ -0,0 +1,3274 @@
...
+            "[NeMo I 2023-01-18 18:51:11 transcribe_speech:293] Writing transcriptions into file: ./test_manifest_predictions.json\n",
+            "[NeMo I 2023-01-18 18:51:11 transcribe_speech:308] Finished writing predictions to ./test_manifest_predictions.json!\n",
+            "[NeMo I 2023-01-18 18:51:11 speech_to_text_eval:108] Finished transcribing speech dataset. Computing ASR metrics..\n",
+            "[NeMo I 2023-01-18 18:51:11 speech_to_text_eval:145] Got WER of 1.3544631306597672\n"
+          ]
+        }
+      ],
+      "source": [
+        "! python ./NeMo/examples/asr/speech_to_text_eval.py \\\n",
+        "    model_path=./Conformer-CTC-BPE/test/checkpoints/Conformer-CTC-BPE.nemo \\\n",
```
Comment (svenchilton) [inline asr-python-finetuning-nemo.ipynb:3081]:
```suggestion
        "    model_path=$MODEL_PATH \\\n",
```

File: asr-python-finetuning-nemo.ipynb [hunk 5]
```diff
@@ -0,0 +1,3274 @@
...
+        "Convert the downloaded model to .riva format, we will use encryption key=nemotoriva. Change this while generating .riva models for production."
+      ],
+      "metadata": {
+        "id": "8XNolXIPptAr"
+      }
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "nemo_file_path = \"./Conformer-CTC-BPE/test/checkpoints/Conformer-CTC-BPE.nemo\"\n",
```
Comment (svenchilton) [inline asr-python-finetuning-nemo.ipynb:3195]:
```suggestion
        "nemo_file_path = MODEL_PATH\n",
```

File: asr-python-finetuning-nemo.ipynb [hunk 6]
```diff
@@ -0,0 +1,3288 @@
...
+    "## Fine-tuning ASR model using NeMo"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "QfRpENJJ_mLy"
+   },
+   "source": [
+    "### Download Data"
```
Comment (svenchilton) [inline asr-python-finetuning-nemo.ipynb:836]:
```suggestion
    "### Download the Data"
```

File: asr-python-finetuning-nemo.ipynb [hunk 7]
```diff
@@ -0,0 +1,3288 @@
...
+    "\n",
+    "\"\"\"Process AN4 dataset.\"\"\"\n",
+    "if not os.path.exists(source_data_dir):\n",
+    "    link = 'http://www.speech.cs.cmu.edu/databases/an4/an4_sphere.tar.gz'\n",
+    "    raise ValueError(\n",
+    "        f\"Data not found at `{source_data_dir}`. Please download the AN4 dataset from `{link}` \"\n",
+    "        f\"and extract it into the folder specified by the `source_data_dir` argument.\"\n",
+    "    )\n",
+    "\n",
+    "# Conversion from SPH files to WAV files\n",
```
Comment (svenchilton) [inline asr-python-finetuning-nemo.ipynb:2170]:
```suggestion
    "# Convert SPH files to WAV files\n",
```

File: asr-python-finetuning-nemo.ipynb [hunk 8]
```diff
@@ -0,0 +1,3288 @@
...
+    "#### Create Tokenizer"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "4zBKAO36_mME"
+   },
+   "source": [
+    "Before we can do the actual training, we need to create a tokenizer as this ASR model uses word-piece encoding. Character based models don't need the tokenizer creation as only single characters are regarded as elements in the vocabulary in their cases. We can use the `process_asr_text_tokenizer.py` script of NeMo to create the tokenizer that generates the subword vocabulary for us for use in training. The size of the vocabulary (vocab_size) should be the same as the vocabulary size as the ASR model. We would clone the NeMo's repository from GitHub to use the scripts and examples available there.\n"
```
Comment (svenchilton) [inline asr-python-finetuning-nemo.ipynb:2240]:
```suggestion
    "Before we can do the actual training, we need to create a tokenizer as this ASR model uses word-piece encoding. Character based models don't need the tokenizer creation as only single characters are regarded as elements in the vocabulary in their cases. We can use NeMo's `process_asr_text_tokenizer.py` script to create the tokenizer that generates the subword vocabulary for us for use in training. The size of the vocabulary (`vocab_size`) should be the same as the vocabulary size in the ASR model. We will clone the NeMo GitHub repository to use the scripts and examples available there.\n"
```

File: asr-python-finetuning-nemo.ipynb [hunk 9]
```diff
@@ -0,0 +1,3288 @@
...
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "6-yDMTMO_mMJ"
+   },
+   "source": [
+    "#### Training Conformer-CTC\n",
+    "\n",
+    "NeMo uses configuartion files to configure the training parameters. You may update them directly by editing the configuration file or from the command-line interface. For example, if the number of epochs are needed to be modified along with a change in the learning rate, you can add `trainer.max_epochs=100` and `optim.lr=0.02` and train the model. \n",
```
Comment (svenchilton) [inline asr-python-finetuning-nemo.ipynb:2363]:
```suggestion
    "NeMo uses configuartion files to configure the training parameters. You may update them directly by editing the configuration file or from the command-line interface. For example, if the number of epochs needs to be modified, along with a change in the learning rate, you can add `trainer.max_epochs=100` and `optim.lr=0.02` and train the model. \n",
```

File: asr-python-finetuning-nemo.ipynb [hunk 10]
```diff
@@ -0,0 +1,3288 @@
...
+      "tcmalloc: large alloc 1541218304 bytes == 0xc4f58000 @  0x7fd2fd25e680 0x7fd2fd27eda2 0x5f714c 0x64d800 0x527022 0x5c4520 0x5f6eb7 0x7fd2db442795 0x7fd2db59aeeb 0x7fd2b494bc45 0x7fd2b4945728 0x7fd2b494d2c9 0x7fd2db5ae9ca 0x7fd2db195f00 0x5f5b39 0x5f6706 0x50ba83 0x570b82 0x569d8a 0x5f60c3 0x56bab6 0x569d8a 0x5f60c3 0x570b82 0x5f5ee6 0x56bab6 0x569d8a 0x50b3a0 0x56cc92 0x569d8a 0x50b3a0\n",
+      "`Trainer.fit` stopped: `max_epochs=1` reached.\n",
+      "Epoch 0: 100% 69/69 [01:25<00:00,  1.24s/it, loss=75.5, v_num=test]\n",
+      "tcmalloc: large alloc 1541218304 bytes == 0xc4f58000 @  0x7fd2fd25e680 0x7fd2fd27eda2 0x5f714c 0x64d800 0x527022 0x5c4520 0x5f6eb7 0x7fd2db442795 0x7fd2db59aeeb 0x7fd2b494bc45 0x7fd2b4945728 0x7fd2b494d2c9 0x7fd2db5ae9ca 0x7fd2db195f00 0x5f5b39 0x5f6706 0x50ba83 0x570b82 0x569d8a 0x5f60c3 0x56bab6 0x569d8a 0x5f60c3 0x570b82 0x5f5ee6 0x56bab6 0x569d8a 0x50b3a0 0x56cc92 0x569d8a 0x50b3a0\n",
+      "\u001b[0m"
+     ]
+    }
+   ],
+   "source": [
+    "# To fully train the model from scratch, you'll need to increase trainer.max_epochs from 1\n",
```
Comment (svenchilton) [inline asr-python-finetuning-nemo.ipynb:2782]:
```suggestion
    "# To fully train the model from scratch, you'll need to increase trainer.max_epochs from 1.\n",
```

File: asr-python-finetuning-nemo.ipynb [hunk 11]
```diff
@@ -0,0 +1,3288 @@
...
+    "# Empirical evidence suggests that around 200 epochs should suffice\n",
```
Comment (svenchilton) [inline asr-python-finetuning-nemo.ipynb:2783]:
```suggestion
    "# Empirical evidence suggests that around 200 epochs should suffice.\n",
```

File: asr-python-finetuning-nemo.ipynb [hunk 12]
```diff
@@ -0,0 +1,3288 @@
...
+    "### ASR Model Export"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "ap5CjiV4_mMT"
+   },
+   "source": [
+    "With NeMo, you can also export your model in a format that can be deployed using NVIDIA RIVA; a highly performant application framework for multi-modal conversational AI services using GPUs. The same command for exporting to ONNX can be used here. The only small variation is the configuration for `export_format` in the spec file."
```
Comment (svenchilton) [inline asr-python-finetuning-nemo.ipynb:3118]:
```suggestion
    "With NeMo, you can also export your model in a format that can be deployed using NVIDIA Riva: a highly performant application framework for multi-modal conversational AI services using GPUs. The same command for exporting to ONNX can be used here. The only small variation is the configuration for `export_format` in the spec file."
```

File: asr-python-finetuning-nemo.ipynb [hunk 13]
```diff
@@ -0,0 +1,3288 @@
...
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "myI104s2pY13"
+   },
+   "source": [
+    "#### Install the packages\n",
```
Comment (svenchilton) [inline asr-python-finetuning-nemo.ipynb:3127]:
```suggestion
    "#### Install the Packages\n",
```

File: asr-python-finetuning-nemo.ipynb [hunk 14]
```diff
@@ -0,0 +1,3288 @@
...
+    "\n",
+    "We will now install the packages NeMo and nemo2riva. nemo2riva is available on [ngc](https://catalog.ngc.nvidia.com/orgs/nvidia/teams/riva/resources/riva_quickstart/files?version=2.8.1). Make sure you install NGC CLI first before running the following commands."
```
Comment (svenchilton) [inline asr-python-finetuning-nemo.ipynb:3129]:
```suggestion
    "We will now install the NeMo and `nemo2riva` packages. `nemo2riva` is available on [NVIDIA NGC](https://catalog.ngc.nvidia.com/orgs/nvidia/teams/riva/resources/riva_quickstart/files?version=2.8.1). Make sure you install NGC CLI first before running the following commands."
```

File: asr-python-finetuning-nemo.ipynb [hunk 15]
```diff
@@ -0,0 +1,3288 @@
...
+    "!pip install protobuf==3.20.0"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "8XNolXIPptAr"
+   },
+   "source": [
+    "#### Convert to RIVA.\n",
```
Comment (svenchilton) [inline asr-python-finetuning-nemo.ipynb:3202]:
```suggestion
    "#### Convert to Riva\n",
```

File: asr-python-finetuning-nemo.ipynb [hunk 16]
```diff
@@ -0,0 +1,3288 @@
...
+    "\n",
+    "Convert the downloaded model to .riva format, we will use encryption key=nemotoriva. Change this while generating .riva models for production."
```
Comment (svenchilton) [inline asr-python-finetuning-nemo.ipynb:3204]:
```suggestion
    "Convert the downloaded model to the `.riva` format. We will set the encryption key with `--key=nemotoriva`. Choose a different encryption key value when generating `.riva` models for production."
```

File: asr-python-finetuning-nemo.ipynb [hunk 17]
```diff
@@ -0,0 +1,3288 @@
...
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "IAtoNe9krC2z"
+   },
+   "source": [
+    "## More Resources\n",
+    "You may find more info and details on working with NeMo's ASR model here in the tutorials here:\n",
```
Comment (svenchilton) [inline asr-python-finetuning-nemo.ipynb:3238]:
```suggestion
    "You can find more information about working with NeMo's ASR models in the [ASR section](https://github.com/NVIDIA/NeMo/tree/main/tutorials/asr) of the NeMo tutorials."
```

File: asr-python-finetuning-nemo.ipynb [hunk 18]
```diff
@@ -0,0 +1,3288 @@
...
+    "\n",
```
Comment (svenchilton) [inline asr-python-finetuning-nemo.ipynb:3239]:
```suggestion
```

File: asr-python-finetuning-nemo.ipynb [hunk 19]
```diff
@@ -0,0 +1,3288 @@
...
+    "[NeMo Tutorials](https://github.com/NVIDIA/NeMo/tree/main/tutorials/asr)"
```
Comment (svenchilton) [inline asr-python-finetuning-nemo.ipynb:3240]:
```suggestion
```

File: asr-python-finetuning-nemo.ipynb [hunk 20]
```diff
@@ -0,0 +1,3288 @@
...
+    "## What's Next?"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "SJ7hWpRT_mMb"
+   },
+   "source": [
+    "You could use NeMo to build custom models for your own applications, and deploy them to Nvidia Riva! To try deploying these models to RIVA, use the text-to-speech-deployment.ipynb as a quick sample."
```
Comment (svenchilton) [inline asr-python-finetuning-nemo.ipynb:3258]:
```suggestion
    "You can use NeMo to build custom models for your own applications, and deploy them with NVIDIA Riva! Refer to the [Conformer-CTC deployment tutorial](https://github.com/nvidia-riva/tutorials/blob/main/asr-deployment-conformer-ctc.ipynb)."
```

**Review by svenchilton — APPROVED** [review]
Attempting to reimplement @LynseyFabel's editing suggestions. I added a few of my own.

**Pull request merged**